nobrainerzoo/cli.py

```diff
@@ -110,20 +110,36 @@ def predict(
     spec_file = model_dir / "spec.yaml"
     
     if not model_dir.exists():
-        raise Exception("model directory not found")
+        raise Exception("model directory not found!",
+                        "This model does not exist in the zoo or didn't properly added.")
     if not spec_file.exists():
-        raise Exception("spec file doesn't exist")
+        raise Exception("spec file doesn't exist!",
+                        "This model does not exist in the zoo or didn't properly added.")
   
     with spec_file.open() as f:
         spec = yaml.safe_load(f)
 
-    image = _container_check(container_type=container_type, image_spec=spec.get("image"),
-                              docker_ok=False)
+    image = _container_check(container_type=container_type, image_spec=spec.get("image"))
     
-    download_image = Path(__file__).resolve().parents[0] / "env/nobrainer-zoo_test.sif"
-    cmd0 = ["singularity", "run", download_image, "python3", "nobrainerzoo/download.py", model]
+    if container_type == "singularity":
+        
+        download_image = Path(__file__).resolve().parents[0] / "env/nobrainer-zoo_nobrainer.sif.sif"
+        if not download_image.exists():
+            dwnld_cmd = ["singularity", "pull", "--dir", 
+                       str(Path(__file__).resolve().parents[0]/ "env"),
+                       "docker://neuronets/nobrainer-zoo:nobrainer"]
+            p = sp.run(dwnld_cmd, stdout=sp.PIPE, stderr=sp.STDOUT, text=True)
+            print(p.stdout)
+        cmd0 = ["singularity", "run", download_image, "python3", "nobrainerzoo/download.py", model]
+    elif container_type == "docker":
+        path = str(Path(__file__).resolve())+":"+str(Path(__file__).resolve())
+        file = str(Path(__file__).resolve() / "download.py")
+        # check output option
+        cmd0 = ["docker", "run","-v",path,"--rm","neuronets/nobrainer-zoo:nobrainer", 
+                "python3", file, model]
+        
```

Comment:
could you raise an error if the container_type is not docker or singularity?

```suggestion
        else:
            raise ValueError(f"unknown container type: {container_type}")
```